src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -82,7 +82,7 @@ internal static bool IsOrleansShallowCopyable(this Type t)
 
         private static bool IsTypeFieldsShallowCopyable(TypeInfo typeInfo)
         {
-            return typeInfo.GetFields().All(f => !(f.FieldType.IsEquivalentTo(typeInfo)) && IsOrleansShallowCopyable(f.FieldType));
+            return typeInfo.GetFields().All(f => !f.FieldType.GetTypeInfo().IsEquivalentTo(typeInfo.AsType()) && IsOrleansShallowCopyable(f.FieldType));
```

Comment:
but what about doing this at least:
`return typeInfo.GetFields().All(f => f.FieldType != typeInfo.AsType()) && IsOrleansShallowCopyable(f.FieldType));`
This way we don't get the TypeInfo unecessarily in the first place